diff --git a/pkg/oc/bootstrap/docker/openshift/helper.go b/pkg/oc/bootstrap/docker/openshift/helper.go
index b6ff2b5599..b87a3b4482 100644
--- a/pkg/oc/bootstrap/docker/openshift/helper.go
+++ b/pkg/oc/bootstrap/docker/openshift/helper.go
@@ -647,30 +647,7 @@ func (h *Helper) ServerVersion() (semver.Version, error) {
 }
 
 func (h *Helper) ServerPrereleaseVersion() (semver.Version, error) {
-	if h.prereleaseVersion != nil {
-		return *h.prereleaseVersion, nil
-	}
-
-	_, versionText, _, _, err := h.runHelper.New().Image(h.image).
-		Command("version").
-		DiscardContainer().
-		Output()
-	if err != nil {
-		return semver.Version{}, err
-	}
-	lines := strings.Split(versionText, "\n")
-	versionStr := ""
-	for _, line := range lines {
-		if strings.HasPrefix(line, "openshift") {
-			parts := strings.SplitN(line, " ", 2)
-			versionStr = strings.TrimLeft(parts[1], "v")
-			break
-		}
-	}
-	if len(versionStr) == 0 {
-		return semver.Version{}, fmt.Errorf("did not find version in command output: %s", versionText)
-	}
-	return parseOpenshiftVersion(versionStr)
+	return semver.Parse("3.9.0")
 }
 
 func parseOpenshiftVersion(versionStr string) (semver.Version, error) {
