diff --git a/pkg/oc/clusterup/up.go b/pkg/oc/clusterup/up.go
index 4093b035bb..afaf8e0e05 100644
--- a/pkg/oc/clusterup/up.go
+++ b/pkg/oc/clusterup/up.go
@@ -310,15 +310,8 @@ func (c *ClusterUpConfig) Complete(cmd *cobra.Command) error {
 		return err
 	}
 
-	// Check whether the Docker host has the right binaries to use Kubernetes' nsenter mounter
-	// If not, use a shared volume to mount volumes on OpenShift
-	if isRedHatDocker, err := c.DockerHelper().IsRedHat(); err == nil && isRedHatDocker {
-		c.printProgress("Checking type of volume mount")
-		c.UseNsenterMount, err = c.HostHelper().CanUseNsenterMounter()
-		if err != nil {
-			return err
-		}
-	}
+	// NixOS: Assume nsenter
+	c.UseNsenterMount = true
 
 	if err := os.MkdirAll(c.BaseDir, 0755); err != nil {
 		return err
